### PR TITLE
Add downstream container files and update the konflux pipeline

### PR DESCRIPTION
**Pull request by msherif1234**

### CODE REVIEW

File: Containerfile.bpfman-agent.openshift [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+# Copy the Go Modules manifests
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+COPY vendor/ vendor/
+
+# cache deps before building and copying source so that we don't need to re-download as much
+# and so that source changes don't invalidate our downloaded layer
+RUN go mod download
```
Comment (frobware) [inline Containerfile.bpfman-agent.openshift:23]: Shouldn't the vendor directory already be up to date? The expectation is that go mod tidy and go mod vendor are run for each commit, ensuring consistency and avoiding the need to download dependencies locally or in CI.

Any commit should trigger a CI check for go mod tidy. If this check reports a difference, the commit is incomplete and should not be merged.

File: Containerfile.bpfman-agent.openshift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ARG TARGETARCH
+ARG TARGETPLATFORM
+
+WORKDIR /
+COPY --from=bpfman-agent-build /usr/src/bpfman-operator/bpfman-agent .
+
+# Install crictl
+RUN dnf -y install wget tar gzip
+ARG VERSION="v1.28.0"
+RUN wget --no-check-certificate https://github.com/kubernetes-sigs/cri-tools/releases/download/${VERSION}/crictl-${VERSION}-linux-${TARGETARCH}.tar.gz
```
Comment (frobware) [inline Containerfile.bpfman-agent.openshift:47]: Why do we need to disable certificate checking, especially for connections to github.com?

File: Containerfile.bpfman-agent.openshift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ARG BUILDPLATFORM=linux/amd64
+
+FROM --platform=$BUILDPLATFORM registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.22-openshift-4.17 AS bpfman-agent-build
+
+# The following ARGs are set internally by docker/build-push-action in github actions
+ARG TARGETOS
+ARG TARGETARCH
+ARG TARGETPLATFORM
+
+RUN echo "TARGETOS=${TARGETOS}  TARGETARCH=${TARGETARCH}  BUILDPLATFORM=${BUILDPLATFORM}  TARGETPLATFORM=${TARGETPLATFORM}"
```
Comment (frobware) [inline Containerfile.bpfman-agent.openshift:11]: nit: there's double spaces between each env key/value pair that is being echoed.

File: Containerfile.bpfman-agent.openshift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+# Copy the go source
+COPY . .
+
+# Build
+# the GOARCH has not a default value to allow the binary be built according to the host where the command
+# was called. For example, if we call make docker-build in a local env which has the Apple Silicon M1 SO
+# the docker BUILDPLATFORM arg will be linux/arm64 when for Apple x86 it will be linux/amd64. Therefore,
+# by leaving it empty we can ensure that the container and binary shipped on it will have the same platform.
+WORKDIR /usr/src/bpfman-operator
+RUN CGO_ENABLED=0 GOOS=${TARGETOS:-linux} GOARCH=${TARGETARCH} go build -mod vendor -a -o bpfman-agent ./cmd/bpfman-agent/main.go
```
Comment (frobware) [inline Containerfile.bpfman-agent.openshift:34]: Do we really need the `-a` flag? We're building in a fresh container, so everything that needs building will get rebuilt regardless. Additionally, CI implicitly ensures a complete build.

**Review by frobware — COMMENTED** [review]
This looks like a copy of an existing containerfile, but I have some concerns regarding the use of go mod vendor during the build. This shouldn't be necessary and would lead to build inconsistencies.

**Comment by msherif1234** [comment]
@frobware I think there is a prob installeting ca-certificate
`Error: building at STEP "RUN ${DNF_CMD} -y install wget tar gzip ca-certificates": exit status 127` I will try with the old way and see if the agent builds

**Review by anfredette — APPROVED** [review]
LGTM

**Pull request merged**